### PR TITLE
Extend file access retries to check rename and remove as well

### PR DESCRIPTION
**Pull request by ehoogeveen-medweb**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

I'm still getting some file access errors on my laptop, so I extended the logic to cover a few more places. I added `try_remove`, matching `try_rename` to avoid bare `os.remove` calls. To avoid code duplication I turned the retry logic into a decorator that wraps `sanitize_open`, `try_remove` and `try_rename`.

Notes:
- `try_rename` had a custom error message. I couldn't add it to the decorator and I think it would have conflicted with the error handling so it now raises the built in error if it fails.
- I added `error.EINVAL` (error 22) to the list of errors to retry, as I also saw this error on my laptop (I'm pretty sure it's the same problem as the parameter looks fine and it doesn't fail every time).

### CODE REVIEW

**Comment by pukkandan** [comment]
self.report_error and raising an error are not the same thing. Will need to investigate whether this might break anything

**Comment by ehoogeveen-medweb** [comment]
Ah, sorry, for some reason I thought `self.report_error` just raised. It's a bit ugly, but I think I could use `func.__name__` to make an exception for `try_rename` and preserve the old behavior.

**Comment by jakeogh** [comment]
@ehoogeveen-medweb are the errors you are encountering happening when there are multiple yt-dlp processes, perhaps attempting to download the same ID?

**Comment by ehoogeveen-medweb** [comment]
@jakeogh No, it happens with only 1 instance of yt-dlp and seems to happen regardless of what the rest of the system is doing (although causing a burst of network activity by e.g. reloading a page seems to make it happen more frequently).

File: yt_dlp/downloader/common.py [hunk 1]
```diff
@@ -210,28 +210,38 @@ def undo_temp_name(self, filename):
...
+            file_access_retries = self.params.get('file_access_retries', 10)
+            retry = 0
+            while True:
+                try:
+                    return func(self, *argv, **kwargs)
+                except (IOError, OSError) as err:
+                    retry = retry + 1
+                    if retry > file_access_retries or err.errno not in (errno.EACCES, errno.EINVAL):
+                        if func.__name__ == 'try_rename':
+                            return self.report_error(f'unable to rename file: {err}')
```
[the remark this answers is not in the document]
Comment (ehoogeveen-medweb) [reply yt_dlp/downloader/common.py]: Ah, adding a parameter to the wrapper function is indeed better, thanks.

**Pull request merged**